Provide a local Environment

brunotag opened the pull request:
environment becomes a module, which provides one of two differentimplementations of the same interface, depending on the value of an environment variable.

tim-hogan commented:
Rather than change classEnvironment.php I would prefer we created a new file and new name and use throughout.  So in effect we create a new class and name (based on classEnvironment.php) that implements the interface.
The reason is that classEnvironment.php comes form another framework and I dont want to end up with two different variations on the same class.  So are you able to create a new class based on that and then we can merge.  You will also need to change classSecurity.php as it has an include.

brunotag commented:
I did do almost exactly what you are suggesting.

The new class is classDevEnvironment.
The old classEnvironment is still there, same logic. I had to make some
changes to it, because it had to implement the new interface formally.
They both implement the interface, so I had to make some changes to
classEnvironment.

-------------------------

The only additional thing I did is to refactor the *instantiation* of the
two classes (classEnvironment / classDevEnvironment) into a module, which
then is referenced by the rest of the code.

With this approach:

   - we get the benefits of modular polymorphism (the "rest" of the code
   only has to include the module, and it gets the correct implementation
   automagically)
   - the two classes, classEnvironment / classDevEnvironment, don't know
   about each other
   - the rest of the code doesn't have to know any implementation detail,
   it only has to know that there's a module that needs to be required.
   - all the "environment selection" logic is hidden in the module


On Sat, 17 Apr 2021 at 11:13, tim-hogan ***@***.***> wrote:

> Rather than change classEnvironment.php I would prefer we created a new
> file and new name and use throughout. So in effect we create a new class
> and name (based on classEnvironment.php) that implements the interface.
> The reason is that classEnvironment.php comes form another framework and I
> dont want to end up with two different variations on the same class. So are
> you able to create a new class based on that and then we can merge. You
> will also need to change classSecurity.php as it has an include.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tim-hogan/glidingops/pull/76#issuecomment-821717836>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJIYBEOA3DVTHZRDZYAZ5DTJDACVANCNFSM43BGFL6A>
> .
>


-- 
Bruno Tagliapietra
***@***.***
0226441495
+64226441495

tim-hogan commented:
We need to change the name of the file classEnvironment, if we are going to edit it as its from another framework and thus different.
So suggest we have classProdEnvironment and classDevEnvrionment.
Also I need to think what we do with classSecure as this also requires changing, whihc is not in this pull request.

tim-hogan commented:
See my notes in github.

Tim

From: Bruno Tagliapietra ***@***.***>
Sent: Monday, 19 April 2021 12:42 am
To: tim-hogan/glidingops ***@***.***>
Cc: Tim Hogan ***@***.***>; Comment ***@***.***>
Subject: Re: [tim-hogan/glidingops] Provide a local Environment (#76)

I did do almost exactly what you are suggesting.

The new class is classDevEnvironment.
The old classEnvironment is still there, same logic. I had to make some
changes to it, because it had to implement the new interface formally.
They both implement the interface, so I had to make some changes to
classEnvironment.

-------------------------

The only additional thing I did is to refactor the *instantiation* of the
two classes (classEnvironment / classDevEnvironment) into a module, which
then is referenced by the rest of the code.

With this approach:

- we get the benefits of modular polymorphism (the "rest" of the code
only has to include the module, and it gets the correct implementation
automagically)
- the two classes, classEnvironment / classDevEnvironment, don't know
about each other
- the rest of the code doesn't have to know any implementation detail,
it only has to know that there's a module that needs to be required.
- all the "environment selection" logic is hidden in the module


On Sat, 17 Apr 2021 at 11:13, tim-hogan ***@***.***<mailto:***@***.***>> wrote:

> Rather than change classEnvironment.php I would prefer we created a new
> file and new name and use throughout. So in effect we create a new class
> and name (based on classEnvironment.php) that implements the interface.
> The reason is that classEnvironment.php comes form another framework and I
> dont want to end up with two different variations on the same class. So are
> you able to create a new class based on that and then we can merge. You
> will also need to change classSecurity.php as it has an include.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tim-hogan/glidingops/pull/76#issuecomment-821717836>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJIYBEOA3DVTHZRDZYAZ5DTJDACVANCNFSM43BGFL6A>
> .
>


--
Bruno Tagliapietra
***@***.***<mailto:***@***.***>
0226441495
+64226441495


—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/tim-hogan/glidingops/pull/76#issuecomment-821985892>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AE2HBA4CCVE3HRZGYV7V47TTJLHSBANCNFSM43BGFL6A>.

brunotag commented:
right, that's easy. I'll do it later.

That Secure class to me looks like it does way too much work.
First thing, I'd say modify it to take away all the bits and pieces we
don't need (rename it if we must).
Maybe we start defining one or more interfaces containing all the pieces of
functionality we would actually need.
Then, if we want the implementation of "Secure" (or a subset of it) to
differ according to different environments, all those static methods must
become instance methods. At least all the public ones.
Make as many as possible private would help.
Then we need a way to inject an instance of "Secure" instead of referencing
the class name. That can be achieved by applying the modularisation pattern.


On Mon, 19 Apr 2021 at 07:30, tim-hogan ***@***.***> wrote:

> We need to change the name of the file classEnvironment, if we are going
> to edit it as its from another framework and thus different.
> So suggest we have classProdEnvironment and classDevEnvrionment.
> Also I need to think what we do with classSecure as this also requires
> changing, whihc is not in this pull request.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tim-hogan/glidingops/pull/76#issuecomment-822047134>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJIYBEXP3XK7ITMZQJVH4LTJMXL3ANCNFSM43BGFL6A>
> .
>


-- 
Bruno Tagliapietra
***@***.***
0226441495
+64226441495

tim-hogan commented:
Secure is quite simple and I think you find we will end up using most as we tighten security,   Again. need to change the name if we are to modify.  Secure should not need to become environment dependent.  The only reason secure has environment is for nthe secure pepper, so am still thinking how we manage this one.

brunotag commented:
Hey, about load_model.php.

Are you happy to get some of the config values from environment variables
and some other ones from devt_environment?
Specifically DATABASE_HOST and DATABASE_PORT.

I'm fine with that, as long as it works: we'll have to ensure that the
variables are created as part of the booting script of the new server.
By the way, are we going to add the booting script to github?



On Tue, 20 Apr 2021 at 06:07, tim-hogan ***@***.***> wrote:

> Secure is quite simple and I think you find we will end up using most as
> we tighten security, Again. need to change the name if we are to modify.
> Secure should not need to become environment dependent. The only reason
> secure has environment is for nthe secure pepper, so am still thinking how
> we manage this one.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tim-hogan/glidingops/pull/76#issuecomment-822668346>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJIYBBJOXNOY5MV4HLWKC3TJRWNRANCNFSM43BGFL6A>
> .
>


-- 
Bruno Tagliapietra
***@***.***
0226441495
+64226441495

brunotag commented:
I have

- renamed classEnvironment to classProdEnvironment
- fixed some problems related with local development in Vagrant
- enabled a phpInfo page that only comes up locally when running in Vagrant, doesn't exist in production
- fixed some problems related with DB access and Laravel, but can't be sure until it runs in production, so we need to merge and deploy

brunotag commented:
Hi Tim,

any thoughts? Merge, not merge, changes...?

On Tue, 20 Apr 2021 at 11:49, Bruno Tagliapietra <
***@***.***> wrote:

> Hey, about load_model.php.
>
> Are you happy to get some of the config values from environment variables
> and some other ones from devt_environment?
> Specifically DATABASE_HOST and DATABASE_PORT.
>
> I'm fine with that, as long as it works: we'll have to ensure that the
> variables are created as part of the booting script of the new server.
> By the way, are we going to add the booting script to github?
>
>
>
> On Tue, 20 Apr 2021 at 06:07, tim-hogan ***@***.***> wrote:
>
>> Secure is quite simple and I think you find we will end up using most as
>> we tighten security, Again. need to change the name if we are to modify.
>> Secure should not need to become environment dependent. The only reason
>> secure has environment is for nthe secure pepper, so am still thinking how
>> we manage this one.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/tim-hogan/glidingops/pull/76#issuecomment-822668346>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAJIYBBJOXNOY5MV4HLWKC3TJRWNRANCNFSM43BGFL6A>
>> .
>>
>
>
> --
> Bruno Tagliapietra
> ***@***.***
> 0226441495
> +64226441495
>


-- 
Bruno Tagliapietra
***@***.***
0226441495
+64226441495

tim-hogan commented:
Not got to it yet.

Soon

From: Bruno Tagliapietra ***@***.***>
Sent: Thursday, 22 April 2021 9:47 pm
To: tim-hogan/glidingops ***@***.***>
Cc: Tim Hogan ***@***.***>; Comment ***@***.***>
Subject: Re: [tim-hogan/glidingops] Provide a local Environment (#76)

Hi Tim,

any thoughts? Merge, not merge, changes...?

On Tue, 20 Apr 2021 at 11:49, Bruno Tagliapietra <
***@***.***<mailto:***@***.***>> wrote:

> Hey, about load_model.php.
>
> Are you happy to get some of the config values from environment variables
> and some other ones from devt_environment?
> Specifically DATABASE_HOST and DATABASE_PORT.
>
> I'm fine with that, as long as it works: we'll have to ensure that the
> variables are created as part of the booting script of the new server.
> By the way, are we going to add the booting script to github?
>
>
>
> On Tue, 20 Apr 2021 at 06:07, tim-hogan ***@***.***<mailto:***@***.***>> wrote:
>
>> Secure is quite simple and I think you find we will end up using most as
>> we tighten security, Again. need to change the name if we are to modify.
>> Secure should not need to become environment dependent. The only reason
>> secure has environment is for nthe secure pepper, so am still thinking how
>> we manage this one.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/tim-hogan/glidingops/pull/76#issuecomment-822668346>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAJIYBBJOXNOY5MV4HLWKC3TJRWNRANCNFSM43BGFL6A>
>> .
>>
>
>
> --
> Bruno Tagliapietra
> ***@***.***<mailto:***@***.***>
> 0226441495
> +64226441495
>


--
Bruno Tagliapietra
***@***.***<mailto:***@***.***>
0226441495
+64226441495


—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/tim-hogan/glidingops/pull/76#issuecomment-824698804>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AE2HBAYZOVCGRNK6EEAJDETTJ7WB7ANCNFSM43BGFL6A>.

tim-hogan commented:
Still missing change to classSecure  ?
What was the rational to extract club messages ?

brunotag commented:
I don't know what changes we want to make to classSecure, sorry.

I extracted the club messages out because that's just standard refactoring
practices by componentisation: it reduces the size of the home page file,
making it more readable, because it pulls out a component (which is also
accidentally reusable) out of it.

Can we implement whatever change we have to introduce in classSecure in a
following pull request later?

On Fri, 23 Apr 2021 at 10:36, tim-hogan ***@***.***> wrote:

> Still missing change to classSecure ?
> What was the rational to extract club messages ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tim-hogan/glidingops/pull/76#issuecomment-825226305>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJIYBHXYPRQRJHM6A5UNBLTKCQHNANCNFSM43BGFL6A>
> .
>


-- 
Bruno Tagliapietra
***@***.***
0226441495
+64226441495

tim-hogan commented:
classSecure includes classEnvrionment, so without that the whole thing is broken.

T

From: Bruno Tagliapietra ***@***.***>
Sent: Friday, 23 April 2021 11:04 am
To: tim-hogan/glidingops ***@***.***>
Cc: Tim Hogan ***@***.***>; Comment ***@***.***>
Subject: Re: [tim-hogan/glidingops] Provide a local Environment (#76)

I don't know what changes we want to make to classSecure, sorry.

I extracted the club messages out because that's just standard refactoring
practices by componentisation: it reduces the size of the home page file,
making it more readable, because it pulls out a component (which is also
accidentally reusable) out of it.

Can we implement whatever change we have to introduce in classSecure in a
following pull request later?

On Fri, 23 Apr 2021 at 10:36, tim-hogan ***@***.***<mailto:***@***.***>> wrote:

> Still missing change to classSecure ?
> What was the rational to extract club messages ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tim-hogan/glidingops/pull/76#issuecomment-825226305>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJIYBHXYPRQRJHM6A5UNBLTKCQHNANCNFSM43BGFL6A>
> .
>


--
Bruno Tagliapietra
***@***.***<mailto:***@***.***>
0226441495
+64226441495


—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/tim-hogan/glidingops/pull/76#issuecomment-825236501>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AE2HBAYMQO2S7ENMITA6YO3TKCTORANCNFSM43BGFL6A>.

brunotag commented:
Done. I don't know how I missed that.
